src/dtoValidation/processors/AdulthoodProcessor.java

```diff
@@ -0,0 +1,37 @@
+package dtoValidation.processors;
+
+import dtoValidation.annotations.Adulthood;
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+
+public class AdulthoodProcessor<T> implements AnnotationProcessor<T> {
+
+    AnnotationProcessor<T> nextProcessor;
```

Comment:
Please make this field as private

Reply:
You defined this field in all processors, and in case of new annotation, you also need to add this field.
What if make AnnotationProcessor as a abstract class and define AnnotationProcessor<T> nextProcessor; field on it, in that case, all subclasses of AnnotationProcessor class already must have that field